### PR TITLE
Add boundary test for long arrays in Array#toSpliced

### PR DESCRIPTION
**Pull request by acutmore**
Test to catch any off by one checks in `toSpliced`. e.g:

```diff
- if (newLen >= Number.MAX_SAFE_INTEGER) {
+ if (newLen > Number.MAX_SAFE_INTEGER) {
      throw new TypeError();
  }
```

### CODE REVIEW

File: test/built-ins/Array/prototype/toSpliced/length-exceeding-array-length-limit.js [hunk 1]
```diff
@@ -28,10 +28,10 @@ var arrayLike = {
     throw new Test262Error("Get 0");
   },
   get "4294967295" () { // 2 ** 32 - 1
-    throw new Test262Error("Get 2147483648");
```
Comment (acutmore) [inline test/built-ins/Array/prototype/toSpliced/length-exceeding-array-length-limit.js:31]: unrelated drive-by cleanups

**Comment by nicolo-ribaudo** [comment]
Thanks!
